agent/engine/task_manager.go

```diff
@@ -543,10 +543,26 @@ func (mtask *managedTask) handleEventError(containerChange dockerContainerChange
 	// event.Status is the desired container transition from container's known status
 	// (* -> event.Status)
 	case api.ContainerPulled:
-		// Container's desired transition was to 'PULLED'. A failure to pull might
-		// not be fatal if e.g. the image already exists.
-		seelog.Errorf("Managed task [%s]: Error while pulling container %s, will try to run anyway: %v",
-			mtask.Arn, container.Name, event.Error)
+		// If the agent pull behavior is always or once, we receive the error because
+		// the image pull fails, the task should fail. If we don't fail task here,
+		// then the cached image will probably be used for creating container, and we
+		// don't want to use cached image for both cases.
+		if mtask.cfg.ImagePullBehavior == config.ImagePullAlwaysBehavior ||
+			mtask.cfg.ImagePullBehavior == config.ImagePullOnceBehavior {
+			seelog.Criticalf("Managed task [%s]: Error while pulling container %s and image %s, task will fail: %v",
```

Comment:
This can be `Errorf` and the message can be `Managed task [%s]: Error while pulling image %s for container %s , moving task to STOPPED:`

Reply:
OK